src/Context/Auth-context/Auth-context.js

```diff
@@ -4,24 +4,55 @@ const authContext = createContext();
 const useAuth = () => useContext(authContext);
 
 const AuthProvider = ({ children }) => {
-  const [userData, setUserData] = useState([]);
+  const [credentialData, setCredentailData] = useState({
+    tokenData: localStorage.getItem("token"),
+    isAuth: localStorage.getItem("token") ? true : false,
+  });
+  const { tokenData, isAuth } = credentialData;
 
-  const loginHandler = async ({ email, password }) => {
+  const signupHandler = async (data) => {
     try {
-      const response = await axios.post(`/api/auth/login`, {
-        email: email,
-        password: password,
+      const response = await axios.post("/api/auth/signup", {
+        email: data.email,
+        password: data.password,
+        firstName: data.firstName,
+        lastName: data.lastName,
       });
       // saving the encodedToken in the localStorage
       localStorage.setItem("token", response.data.encodedToken);
-      setUserData(response.data.encodedToken);
     } catch (error) {
-      console.log(error);
+      console.error(error);
+    }
+  };
+  const loginHandler = async ({ email, password }) => {
+    if (isAuth) {
+      try {
+        const response = await axios.post("/api/auth/login", {
+          email: email,
+          password: password,
+        });
```

Comment:
when key and value pair are same, then you can mention key only. as I mentioned in the above code

Reply:
okk this is somthing new for me.